sdk/eventhub/azure-eventhub/azure/eventhub/exceptions.py

```diff
@@ -27,7 +27,15 @@ def __init__(self, message: str, details: Optional[Union[List[str], "AMQPExcepti
         self.details: Union[List[str], "AMQPException"]
         if details and isinstance(details, Exception):
             self.details = details
-            if isinstance(self.details, AMQPException):
+            if isinstance(self.details, list):
+                try:
+                    details.description = cast(str, details.description)
+                    self._parse_error(details.description)
+                    for detail in self.details:
+                        self.message += "\n{}".format(detail)
+                except:  # pylint: disable=bare-except
+                    self.message += "\n{}".format(details)
+            else:
```

Comment:
if we say else, this will do this for all exceptions, not just AMQPExceptions. Do we want that? Will all other errors have details/condition?